### ISSUE TITLE
Exclusion of Inactive, Disabled, Support, and Service Users from Sharing Dialogs

### ISSUE THREAD

**Issue by dj4oC** [owncloud/ocis#8581]
**As** a normal user of ownCloud Infinite Scale,
**I want** to only see active and regular users in the sharing dialogs,
**So that** I can easily find and share with the relevant people without being overwhelmed by inactive, disabled, support, and service accounts.

**Acceptance Criteria:**

1. When I access the sharing dialog, I should only see a list of active and regular users.
2. Any user account that is labeled as 'inactive', 'disabled', 'support', or 'service' or is stored in another realm should not appear in the sharing dialog list.
3. The search functionality within the sharing dialog should also filter out the aforementioned types of accounts, returning only active and regular users in the search results.

/cc @tbsbdr @wkloucek @micbar @JammingBen

**Comment by rhafer** [comment]
related ticket: https://github.com/owncloud/ocis/issues/8161

Also: Last time a I asked the about this (IIRC @tbsbdr and @micbar were involved), it was agreed that the  "disable user" feature was just meant for being able to prevent user's from signing in. Excluding them from user listings and sharing was specifically not wanted.

**Comment by tbsbdr** [comment]
> Excluding them from user listings and sharing was specifically not wanted.
correct, that is how we wanted it to be for user with the status "**login disallowed**"
